### ISSUE TITLE
Is this the correct way to turn off gradient in the MeanShift layer?

### ISSUE THREAD

**Issue by Teyen-Forde** [thstkdgus35/EDSR-PyTorch#80]
https://github.com/thstkdgus35/EDSR-PyTorch/blob/70294bb6170651dd1c2435da18a34c0e2cf63511/src/model/common.py#L21
nn.Module or nn.Conv2d does not have the "requires_grad" attribute.

I think the gradient should be turned off in this way:
self.weight.requires_grad = False
self.bias.requires_grad = False

In my experiments, I found the weights and bias in MeasShift layer have been modified, meaning they were also learned during training.

Could you clarify this issue?

Thanks,

**Comment by thstkdgus35** [comment]
Hello.

Thank you for reporting the bug.

I have checked some pre-trained models, but there was no problem with sub_mean layers.

However, as you mentioned, ``requires_grad`` option seems not to work well.

Although it was not intended, I think it does not affect the performance since the model will learn optimal sub_mean values.

Anyway, I will fix it soon.

Thank you!

**Comment by Teyen-Forde** [comment]
Thanks for confirming it.
I had modified the MeanShift class into following (if it may be helpful to you):
Now the rgb_mean and rgb_std would not change during training since they are buffers rather than Parameters.
```
class MeanShift(nn.Module):
    def __init__(
        self, rgb_range,
        rgb_mean=(0.4488, 0.4371, 0.4040), rgb_std=(1.0, 1.0, 1.0), sign=-1):

        super(MeanShift, self).__init__()
        self.register_buffer('rgb_mean', torch.zeros(3))
        self.register_buffer('rgb_std', torch.ones(3))
        self.rgb_mean =  rgb_range * torch.Tensor(rgb_mean) 
        self.rgb_std = torch.Tensor(rgb_std)
        if sign < 0:
            self.rgb_mean = -self.rgb_mean/self.rgb_std
            self.rgb_std = 1/self.rgb_std
    def forward(self,x):
        mean = x.new(self.rgb_mean).view(1,3,1,1).expand_as(x)
        std = x.new(self.rgb_std).view(1,3,1,1).expand_as(x)
        x.mul_(std).add_(mean)
        return x
```
